playwright-junit/src/test/java/com/microsoft/playwright/junit/BrowserCustomFactoryTests.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.playwright.junit;
+
+import com.microsoft.playwright.Browser;
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@UseBrowserFactory(CustomBrowserFactory.class)
```

Comment:
I wonder if it'd be more appropriate to make this a parameter/field annotation. It'd be probably more verbose but also more explicit, what are you thoughts?

Reply:
Are you thinking something like this:

```java
@Test
void foo(@UseBrowserFactory(CustomBrowser.class) Page page) {

  //test

}
```

Personally I think class level annotation is good as it is saying the at all tests in this class will use this browser factory.

We can allow further customizing in future PRs.

Reply:
Yes, I meant something like @TempDir junit annotation. I agree that having one class-level annotation is more convenient, let's stick with that, can definitely be customized in the future.

Reply:
One thing that is not very clear with class level annotation is fields with type `Browser`, but it's probably fine if they are not assigned by the fixture.